File: solutions/ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(i)
    s = "#{"Pling" if i % 3 == 0}#{"Plang" if i % 5 == 0}#{"Plong" if i % 7 == 0}"
    s.empty? ? i.to_s : s
  end
end

File: solutions/r/raindrops/raindrops.R
raindrops <- function(number) {
  i <- if (number %% 3 == 0) "Pling" else ""
  a <- if (number %% 5 == 0) "Plang" else "" 
  o <- if (number %% 7 == 0) "Plong" else ""
  if (i == "" && a == "" && o == "") as.character(number)
  else sprintf("%s%s%s", i, a, o)
}
